ansible/roles/test/files/ptftests/vxlan-decap.py

```diff
@@ -30,6 +31,54 @@
 from pprint import pprint
 from pprint import pformat
 
+def count_matched_packets_helper(test, exp_packet, exp_packet_number, port, device_number=0, timeout=1):
+    """
+    Add exp_packet_number to original ptf interface in order to
+    stop waiting when expected number of packets is received
+    """
+    if timeout <= 0:
+        raise Exception("%s() requires positive timeout value." % sys._getframe().f_code.co_name)
+
+    total_rcv_pkt_cnt = 0
+    while True:
+        result = dp_poll(test, device_number=device_number, port_number=port, timeout=timeout)
```

Comment:
if someone will send you one packet in timeout time, this function will wait exp_packet_number * timeout seconds.

Reply:
Yes, but the original **count_matched_packets** interface has the  exactly same behaviour. Since we send packets continuously in a for loop, I think we may ignore this scenario for now.
And the **count_matched_packets_helper** is an enhanced version, which means it returns faster when all expected packets arrive without waiting for a meanless timeout.